pkg/rpcclient/wsclient.go

```diff
@@ -876,37 +866,57 @@ func (c *WSClient) ReceiveNotaryRequests(flt *neorpc.TxFilter, rcvr chan<- *resu
 	return c.performSubscription(params, r)
 }
 
-// Unsubscribe removes subscription for the given event stream.
+// Unsubscribe removes subscription for the given event stream. It will return an
+// error in case if there's no subscription with the provided ID. Call to Unsubscribe
+// doesn't block notifications receive process for given subscriber, thus, ensure
+// that subscriber channel is properly drained while unsubscription is being
+// performed. You may probably need to run unsubscription process in a separate
+// routine (in parallel with notification receiver routine) to avoid Client's
+// notification dispatcher blocking.
 func (c *WSClient) Unsubscribe(id string) error {
-	return c.performUnsubscription(id)
+	return c.performUnsubscription(id, true)
 }
 
-// UnsubscribeAll removes all active subscriptions of the current client.
+// UnsubscribeAll removes all active subscriptions of the current client. It locs
+// the list of subscribers, so that any other notification received from the server
+// during unsubscription process will completely block this process, thus, use
+// this method carefully.
 func (c *WSClient) UnsubscribeAll() error {
 	c.subscriptionsLock.Lock()
 	defer c.subscriptionsLock.Unlock()
```

Comment:
I'd rather just take a copy of `c.subscriptions` and leave finer-grained locking to `c.performUnsubscription`. It's network communication that can take any amount of time.

Reply:
Fixed.

Reply:
With deferred unlock?

Reply:
I've overlooked it, shuldn't be there, fixed.